packages/circle-demo-webapp/app/routes/wallets.$id/components/NewWalletForm.tsx

```diff
@@ -23,14 +24,13 @@ export function NewWalletForm({ walletSetId, onSuccess }: NewWalletFormProps) {
     const description = formData.get('description') as string;
     const blockchain = formData.get('blockchain') as string;
 
-    const success = await createWallet({ walletSetId, name, description, blockchain });
-
-    if (!success) {
+    const wallet = await createWallet({ walletSetId, name, description, blockchain });
+    if (!wallet) {
       return;
     }
 
     if (typeof onSuccess === 'function') {
-      onSuccess();
+      onSuccess(wallet as Wallet);
```

Comment:
I think createWallet returns a wallet of a type Wallet, so no casting needed here, (if you remove unused boolean from return types)

Reply:
I actually moved this logic to the NewWalletDialog component in my latest PR so there is a conflict here, I'll try to help.